From cf45b09dfc56850cb5372793800264ea00edeeed Mon Sep 17 00:00:00 2001
From: Hannu Lyytinen <hannu.lyytinen@unikie.com>
Date: Sat, 31 Jan 2026 09:29:00 +0200
Subject: [PATCH 3/3] arm64: configs: Enable seL4 hypervisor platform drivers

Add CONFIG_SEL4_PLATFORM=y and CONFIG_SEL4_HYP_FTRACE=y to
defconfig and tegra_prod_defconfig to enable the seL4 hypervisor
ftrace control driver when running as a guest under seL4.

Signed-off-by: Technology Innovation Institute
Upstream-Status: Inappropriate [TII-specific changes]
---
 arch/arm64/configs/defconfig            | 4 ++++
 arch/arm64/configs/tegra_prod_defconfig | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/arch/arm64/configs/defconfig b/arch/arm64/configs/defconfig
index 6a30dda73927..f3f2103b2072 100644
--- a/arch/arm64/configs/defconfig
+++ b/arch/arm64/configs/defconfig
@@ -1343,3 +1343,7 @@ CONFIG_FUNCTION_TRACER=y
 # CONFIG_FUNCTION_GRAPH_TRACER is not set
 # CONFIG_DYNAMIC_FTRACE is not set
 CONFIG_MEMTEST=y
+
+# seL4 Hypervisor Platform
+CONFIG_SEL4_PLATFORM=y
+CONFIG_SEL4_HYP_FTRACE=y
diff --git a/arch/arm64/configs/tegra_prod_defconfig b/arch/arm64/configs/tegra_prod_defconfig
index 47faccd896b6..0cf4130955c2 100644
--- a/arch/arm64/configs/tegra_prod_defconfig
+++ b/arch/arm64/configs/tegra_prod_defconfig
@@ -1209,3 +1209,7 @@ CONFIG_DEBUG_FS=y
 # CONFIG_FTRACE is not set
 CONFIG_IO_STRICT_DEVMEM=y
 CONFIG_MEMTEST=y
+
+# seL4 Hypervisor Platform
+CONFIG_SEL4_PLATFORM=y
+CONFIG_SEL4_HYP_FTRACE=y
-- 
2.43.0

